server/routes/strategies.ts

```diff
@@ -0,0 +1,148 @@
+import { URL } from "url";
+import * as passport from "passport";
+import { Strategy as OAuthStrategy } from "passport-oauth2";
+import * as dotenv from "dotenv"
+import * as requests from "request"
+import { Request } from "express";
+import { createNew, IUser, User } from "../schema";
+
+dotenv.config()
+
+type PassportDone = (err: Error | null, user?: IUser | false, errMessage?: { message: string }) => void;
+type PassportProfileDone = (err: Error | null, profile?: IProfile) => void;
+
+interface IStrategyOptions {
+	passReqToCallback: true; // Forced to true for our usecase
+}
+
+interface IOAuthStrategyOptions extends IStrategyOptions {
+	authorizationURL: string;
+	tokenURL: string;
+	clientID: string;
+	clientSecret: string;
+}
+
+interface IProfile {
+	uuid: string;
+	name: string;
+	email: string;
+	token: string;
+}
+
+export type AuthenticateOptions = passport.AuthenticateOptions & {
+	callbackURL: string;
+};
+
+export class GroundTruthStrategy extends OAuthStrategy {
```

Comment:
this class has a lot of inconsistent and/or too many spaces per indent on each line (some lines appear to have 8 space indents?)